packages/client/src/object/convertWireToOsdkObjects.ts

```diff
@@ -38,55 +38,82 @@ const OriginClient = Symbol();
 const UnderlyingObject = Symbol();
 const InterfaceDefinitions = Symbol();
 
+class LinkFetcherProxyHandler<Q extends AugmentedObjectTypeDefinition<any, any>>
+  implements ProxyHandler<any>
+{
+  constructor(
+    private readonly objDef: Q,
+    private readonly primaryKey: any,
+    private readonly client: MinimalClient,
+  ) {}
+
+  get(_target: any, p: string | symbol) {
+    const linkDef = this.objDef.links[p as string];
+    if (linkDef == null) {
+      return;
+    }
+
+    const objectSet = createObjectSet(this.objDef, this.client).where({
+      [this.objDef.primaryKeyApiName]: this.primaryKey,
+    } as WhereClause<Q>).pivotTo(p as string);
+
+    if (!linkDef.multiplicity) {
+      return {
+        get: <A extends SelectArg<any>>(options?: A) =>
+          fetchSingle(
+            this.client,
+            this.objDef,
+            options ?? {},
+            getWireObjectSet(objectSet),
+          ),
+      };
+    } else {
+      return objectSet;
+    }
+  }
+
+  ownKeys(): ArrayLike<string | symbol> {
+    return Object.keys(this.objDef.links);
+  }
+
+  getOwnPropertyDescriptor(
+    target: any,
+    p: string | symbol,
+  ): PropertyDescriptor | undefined {
+    return {
+      enumerable: true,
```

Comment:
I forget does this need to be `this.objDef.links != null`

Reply:
I decided to drop that for now. we can always add back